Plugins/PluginsShared/PluginUtils.swift

```diff
@@ -0,0 +1,204 @@
+///   - outputDirectory: The directory in which generated source files are created.
+/// - Returns: The constructed arguments to be passed to `protoc` when invoking the `proto-gen-swift` `protoc` plugin.
+func constructProtocGenSwiftArguments(
+  config: CommonConfiguration,
+  using fileNaming: CommonConfiguration.FileNaming?,
+  inputFiles: [URL],
+  protoDirectoryPaths: [URL],
+  protocGenSwiftPath: URL,
+  outputDirectory: URL
+) -> [String] {
+  // Construct the `protoc` arguments.
+  var protocArgs = [
+    "--plugin=protoc-gen-swift=\(protocGenSwiftPath.relativePath)",
+    "--swift_out=\(outputDirectory.relativePath)",
+  ]
+
+  // Add the visibility if it was set
+  if let visibility = config.visibility {
+    protocArgs.append("--swift_opt=Visibility=\(visibility.rawValue)")
+  }
+
+  // Add the file naming
+  if let fileNaming = fileNaming {
+    protocArgs.append("--swift_opt=FileNaming=\(fileNaming.rawValue)")
+  }
+
+  // TODO: Don't currently support implementation only imports
+  //  // Add the implementation only imports flag if it was set
+  //  if let implementationOnlyImports = config.implementationOnlyImports {
+  //      protocArgs.append("--swift_opt=ImplementationOnlyImports=\(implementationOnlyImports)")
+  //  }
+
+  // Add the useAccessLevelOnImports only imports flag if it was set
+  if let useAccessLevelOnImports = config.useAccessLevelOnImports {
+    protocArgs.append("--swift_opt=UseAccessLevelOnImports=\(useAccessLevelOnImports)")
+  }
+
+  protocArgs.append(contentsOf: protoDirectoryPaths.map { "--proto_path=\($0.relativePath)" })
+
+  protocArgs.append(contentsOf: inputFiles.map { $0.relativePath })
+
+  return protocArgs
+}
+
+/// Construct the arguments to be passed to `protoc` when invoking the `proto-gen-grpc-swift` `protoc` plugin.
+/// - Parameters:
+///   - config: The configuration for this operation.
+///   - fileNaming: The file naming scheme to be used.
+///   - inputFiles: The input `.proto` files.
+///   - protoDirectoryPaths: The directories in which `protoc` will search for imports.
+///   - protocGenGRPCSwiftPath: The path to the `proto-gen-grpc-swift` `protoc` plugin.
+///   - outputDirectory: The directory in which generated source files are created.
+/// - Returns: The constructed arguments to be passed to `protoc` when invoking the `proto-gen-grpc-swift` `protoc` plugin.
+func constructProtocGenGRPCSwiftArguments(
+  config: CommonConfiguration,
+  using fileNaming: CommonConfiguration.FileNaming?,
+  inputFiles: [URL],
+  protoDirectoryPaths: [URL],
+  protocGenGRPCSwiftPath: URL,
+  outputDirectory: URL
+) -> [String] {
+  // Construct the `protoc` arguments.
+  var protocArgs = [
+    "--plugin=protoc-gen-grpc-swift=\(protocGenGRPCSwiftPath.relativePath)",
+    "--grpc-swift_out=\(outputDirectory.relativePath)",
+  ]
+
+  if let importPaths = config.importPaths {
+    for path in importPaths {
+      protocArgs.append("-I")
```

Comment:
`-I` is a synonym for `--proto_path` which we also set below, why do we have both and have two source of values for them? Also, the protoc-gen-swift args _only_ have `--proto_path` it seems. Somethin' ain't right here!

Reply:
The two sources of values is because one is user-specified and one is a default one which we add based on the path to the proto file we're currently processing. I've made them both use `--proto_path` and fixed up the missing source.